wsmaster/che-core-api-workspace-shared/src/main/java/org/eclipse/che/api/workspace/shared/Constants.java

```diff
@@ -162,5 +162,14 @@
 
   public static final String NO_ENVIRONMENT_RECIPE_TYPE = "no-environment";
 
+  /** Attribute to mark source of the container. */
+  public static final String CONTAINER_SOURCE_ATTRIBUTE = "source";
+
+  /** Mark containers applied to workspace with help recipe definition. */
+  public static final String RECIPE = "recipe";
```

Comment:
I think the constant name should be more self-explained, like `RECIPE_CONTAINER_SOURCE`. The same about `TOOL` constant

Reply:
Done